jgrapht-core/src/main/java/org/jgrapht/traverse/CrossComponentIterator.java

```diff
@@ -90,34 +97,57 @@ public CrossComponentIterator(Graph<V, E> g)
      */
     public CrossComponentIterator(Graph<V, E> g, V startVertex)
     {
+        this(g, startVertex==null?null:Collections.singletonList(startVertex));
+    }
+
+    /**
+     * Creates a new iterator for the specified graph. Iteration will start at the specified start
+     * vertices. If the specified start vertices is <code>
+     * null</code>, Iteration will start at an arbitrary graph vertex.
+     *
+     * @param g the graph to be iterated.
+     * @param startVertices the vertices iteration to be started.
+     *
+     * @throws IllegalArgumentException if <code>g==null</code> or does not contain
+     *         <code>currentStartVertex</code>
+     */
+    public CrossComponentIterator(Graph<V, E> g, Iterable<V> startVertices)
+    {
         super(g);
 
         /*
-         * Initialize start vertex
+         * Initialize crossComponentTraversal and test for containment
          */
-        this.startVertexIterator = graph.vertexSet().iterator();
-        if (startVertex == null) {
+        this.entireGraphVertexIterator = graph.vertexSet().iterator();
+        if (startVertices == null) {
             this.crossComponentTraversal = true;
-            // pick a start vertex if graph not empty
-            if (startVertexIterator.hasNext()) {
-                this.startVertex = startVertexIterator.next();
-            } else {
-                this.startVertex = null;
-            }
         } else {
             this.crossComponentTraversal = false;
-            if (graph.containsVertex(startVertex)) {
-                this.startVertex = startVertex;
-            } else {
-                throw new IllegalArgumentException("graph must contain the start vertex");
+            for (V v : startVertices) {
+                if (!graph.containsVertex(v)) {
+                    throw new IllegalArgumentException("graph must contain the start vertex");
+                }
             }
+            this.startVertexIterator = startVertices.iterator();
+        }
+
+        /*
+         * Initialize start vertex
+         */
+        Iterator<V> it = crossComponentTraversal?entireGraphVertexIterator:startVertexIterator;
+        // pick a start vertex if possible
+        if (it.hasNext()) {
+            this.currentStartVertex = it.next();
+        } else {
+            this.currentStartVertex = null;
         }
+
     }
 
     @Override
     public boolean hasNext()
```

Comment:
Apart from the change with the exception if the vertex does not belong to the graph, this method must be identical to the original.